koku/api/metrics/views.py

```diff
@@ -36,23 +44,42 @@ class CostModelMetricsMapViewSet(mixins.ListModelMixin, viewsets.GenericViewSet)
 
     """
 
-    queryset = CostModelMetricsMap.objects.all()
     serializer_class = CostModelMetricMapSerializer
     permission_classes = (AllowAny,)
 
     def get_queryset(self):
-        """Get a queryset.
+        """
+        Return the Cost Model Metric Map data.
 
-        Restricts the returned data to provider_uuid if supplied as a query parameter.
+        Filter on source_type
         """
-        queryset = CostModelMetricsMap.objects.all()
         source_type = self.request.query_params.get("source_type")
-        if source_type:
-            queryset = queryset.filter(source_type=source_type)
+        cost_model_metric_map_copy = copy.deepcopy(metric_constants.cost_model_metric_map)
+        try:
+            if source_type:
+                # Filter on source type
+                cost_model_metric_map_copy = list(
+                    filter(lambda x: x.get("source_type") == source_type, cost_model_metric_map_copy)
+                )
+            # Convert source_type to human readable.
+            for metric_map in cost_model_metric_map_copy:
+                metric_map["source_type"] = metric_constants.SOURCE_TYPE_MAP[metric_map["source_type"]]
+        except KeyError:
+            LOG.debug("Malformed JSON", exc_error=True)
```

Comment:
Feels like this should be surfaced if its happening, since this is an in code constant now.
```suggestion
            LOG.error("Malformed JSON", exc_error=True)
```